server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -338,7 +338,7 @@ public boolean isControllerRegistrationSupported() {
 
     public short fetchRequestVersion() {
         if (this.isAtLeast(IBP_3_5_IV1)) {
-            return 15;
+            return 16;
```

Comment:
Hmm. Is this correct? In the upgrade scenario we will send request version 16 to brokers that may not have that version yet. I know we just ignore tagged fields, but I'm not sure I recall if we can handle version bumps.

If this is always just the latest version, should it be hardcoded?

Reply:
I think you're right that this could cause issues during upgrade, I think using the latest version should be safe.

Reply:
I'm just trying to figure out if we would see unsupported version errors during upgrades. I think we might.

Reply:
Clusters with ibp 3.5 were guaranteed to support version 15 since that was the version we defined the ibp. I don't think we can just change the version because some clusters will have ibp 3.5, but not version 16 fetch requests. We could avoid this with the tagged fields, but since we are bumping the version, we run into a problem.

Reply:
Ah I hadn't thought this through. I think this would need an IBP version bump to avoid errors during upgrades?

Reply:
I believe so.

Reply:
I don't think there's any reason to bump the fetch version for replication fetches. this would be purely a consumer client optimization.

Reply:
It is a bit weird to not bump the version though. I guess if there is ever a v17 that is for replication, we would go straight to that (with MV/IBP bump) and that would be ok.

Reply:
I'll remove this version bump.

Reply:
I remember why I added this now, we have a test that checks we're using the latest version for various API's. I guess we would need to remove that for FETCH.
https://github.com/apache/kafka/blob/13b2edd9af802d7db99815e834b062aafc3e2313/core/src/test/scala/unit/kafka/server/ReplicaFetcherThreadTest.scala#L120

Reply:
I chatted with David J about this offline. Since we are using MV/IBP bumps for fetches, a simple thing to do would be to pick up the newest MV for this release and include the fetch bump here.

The alternative is setting up the fetch path to use ApiVersions to ensure the correct version. But that might be out of scope for this change. 

With either of these approaches we can keep the latest version for the replication fetches which would make things a little clearer.

Reply:
> Since we are using MV/IBP bumps for fetches, a simple thing to do would be to pick up the newest MV for this release and include the fetch bump here

ok, so my understanding is instead of bumping IBP to `IBP_3_7_IV1` we would wait for the release of 'IBP_3_8_IV0' to bump the fetch version here to 16.

> The alternative is setting up the fetch path to use ApiVersions to ensure the correct version. But that might be out of scope for this change.

yeah, I would need to look more into this, I'm not familiar enough to know how it might look.

Reply:
> > Since we are using MV/IBP bumps for fetches, a simple thing to do would be to pick up the newest MV for this release and include the fetch bump here

> ok, so my understanding is instead of bumping IBP to IBP_3_7_IV1 we would wait for the release of IBP_3_8_IV0 to bump the fetch version here to 16.

I should have made this a question. I confused myself with the release versions, I think we can use IBP_3_7_IV0 to bump the fetch version to 16 since it hasn't been released yet, let me know if it makes sense to you.